### PR TITLE
Bug fix : kaggle API credential import failed in text classification with tf-hub on kaggle notebook

### PR DESCRIPTION
**Pull request by Tahsin-Mayeesha**
In the function ```get_kaggle_credentials``` when we were writing to the token_file, uploaded['kaggle.json'] returned byte code. In the original notebook, we didn't open the file in binary mode so we didn't write the credentials from the uploaded kaggle.json file to the destination file and the import was showing error that it couldn't find the username in the token_file. 

So we just had to add open(token_file,"wb") to open the file in binary mode. After the fix the notebook works as expected. Here's a copy notebook with the mini fix in action. 

https://colab.research.google.com/drive/1qTiQydX0xvdwD17I0ZtJOCAHXPwPesTs

Fixed code : 
```
import os

# Upload the API token.
def get_kaggle_credentials():
  token_dir = os.path.join(os.path.expanduser("~"),".kaggle")
  token_file = os.path.join(token_dir, "kaggle.json")
 .......
  uploaded = files.upload()
  with open(token_file, "wb") as f:
     f.write(uploaded["kaggle.json"])
  os.chmod(token_file, 600)

get_kaggle_credentials()
# Note: Only import kaggle after adding the credentials.
import kaggle
```

### CODE REVIEW

**Comment by Tahsin-Mayeesha** [comment]
I signed it!

**Comment by Tahsin-Mayeesha** [comment]
> I am not sure if reading this correctly, but does the commit move the indentation left by 1 space? ?>Could you please confirm and if yes, move it back, so that we can get a readable diff?

image


I only changed one line, but removed the jupyter checkpoints when committing. I'm not sure why the model diff is that large. I opened the jupyter notebook in windows, could this be the issue? 

This is the forked versions notebook, in github and in my local machine the rendering is the same as before  : 
https://github.com/TahsinMayeesha/hub/blob/master/examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb

**Comment by Tahsin-Mayeesha** [comment]
@vbardiovskyg I reverted the previous commit and ended up only modifying one line, can you please check? I've not been able to solve where the real issue was though that made such a large diff. I think it was because I didn't removed kernel outputs before committing.

File: examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb [hunk 1]
```diff
@@ -170,7 +170,7 @@
         "      raise no_file\n",
```
Comment (vbardiovskyg) [inline examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb]: Out of place comment:

Could you please try removing the block with "pip installs" and see whether it still works? I think all of the packages should be already running in the colab.

https://github.com/tensorflow/hub/blob/master/examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb?short_path=2e9a14e#L95

File: examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb [hunk 2]
```diff
@@ -170,7 +170,7 @@
...
         "    \n",
```
Comment (vbardiovskyg) [inline examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb]: One thing that happened to me when running this colab. If one calls get_kaggle_credentials() and uploads the wrong file, they can never recover with this through the colab.

Could you please change the function to do something like:

def get_kaggle_credentials(force_upload=False)

Then inside the function if force_upload==True, delete the token_file if it existed and upload a new one.

**Comment by Tahsin-Mayeesha** [comment]
> Thanks for the change, I added two more comments that would make the colab a bit cleaner and easier to use, since you are already working on improving it.

I'll get to it.

File: examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb (hunk 1 above)
Comment (Tahsin-Mayeesha) [reply examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb]: It works without the pip installs.

File: examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb (hunk 2 above)
Comment (Tahsin-Mayeesha) [reply examples/colab/text_classification_with_tf_hub_on_kaggle.ipynb]: I changed the get_kaggle_credential function to do force upload.

**Comment by vbardiovskyg** [comment]
Thanks for the fixes, but now the indentation is moved to the left again. Is that your environment doing that?

**Comment by Tahsin-Mayeesha** [comment]
> Thanks for the fixes, but now the indentation is moved to the left again. Is that your environment doing that?

In the first commit after I deleted the installation cell the other cells shifted up and created a large diff, but the second one seemed okay. It's probably my environment, any idea what I should do to fix it? I've set git core.autocrlf to true as advised here. https://help.github.com/en/articles/dealing-with-line-endings#global-settings-for-line-endings

**Comment by vbardiovskyg** [comment]
Not sure what you are using for editing, but this is what I would do:
1. To change the colab, open the colab in colab.research.com.
2. Make my edits.
3. Download the file to local machine and replace the raw file in the local git branch on my machine.
4. Check for any unintended diffs with "git diff origin"

**Comment by Tahsin-Mayeesha** [comment]
I'm using jupyter notebook with anaconda.

**Comment by vbardiovskyg** [comment]
Can you please remove the last two cells that were added?

**Comment by Tahsin-Mayeesha** [comment]
> Can you please remove the last two cells that were added?

The one's with confusion matrix and the prediction submission? Those were in the original file though.

**Comment by vbardiovskyg** [comment]
Submission cell was there as text. Now it is there as text and then again as code and then there is a new code cell. Or at least that's what "view file" shows me.

**Comment by Tahsin-Mayeesha** [comment]
> Submission cell was there as text. Now it is there as text and then again as code and then there is a new code cell. Or at least that's what "view file" shows me.

Oh yeah, sorry, removing those.

**Comment by vbardiovskyg** [comment]
Looks good now, thanks for the change!

**Comment by Tahsin-Mayeesha** [comment]
> Looks good now, thanks for the change!

thanks, can we merge it now? Also thanks for the fix about the indentation issue.

**Comment by vbardiovskyg** [comment]
Merged into master in a79a81d6cff7aada1f642978e022aadad35fd8aa.

**Pull request closed without merging**